### PR TITLE
fix: Modify DS Get cmd client API to use pointer as return value

### LINKED ISSUE #590: [V2] Http client lib should check whether the http response is empty
Since we have core-command V2 API  which can execute Get command with **ds-returnevent** parameter, so the DS will return **empty content** when ds-returnevent set to **no**. 

And the empty content will cause json unmarshal error, so the Http client lib should check whether the http response is empty.

https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/core-command/2.x#/default/get_device_name__name___command_

### PR DESCRIPTION
**Pull request by weichou1229**
Close #590

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Modify DS Get cmd client API to use the pointer as return type. Then we can use the nil pointer to indicate that the HTTP response content is empty.

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]
Get Command might return CBOR.
Please check the response header and decide how to deal with it.

File: v2/clients/http/utils/request.go [hunk 1]
```diff
@@ -35,6 +37,36 @@ func GetRequest(ctx context.Context, returnValuePointer interface{}, baseUrl str
 	return nil
 }
 
+// GetRequestAndReturnBinaryRes makes the get request and return the binary response
```
Comment (cloudxxx8) [inline v2/clients/http/utils/request.go:40]: the go doc should state all the returned value, include the contentType

**Review by cloudxxx8 — CHANGES_REQUESTED** [review]

File: v2/clients/http/utils/request.go (hunk 1 above)
Comment (weichou1229) [reply v2/clients/http/utils/request.go]: Modified.

**Review by cloudxxx8 — APPROVED** [review]
LGTM

**Pull request merged**